substrate/frame/collective/src/lib.rs

```diff
@@ -648,6 +819,59 @@ pub mod pallet {
 
 			Self::do_close(proposal_hash, index, proposal_weight_bound, length_bound)
 		}
+
+		/// Disapprove the proposal and burn the cost held for storing this proposal.
+		///
+		/// Parameters:
+		/// - `origin`: must be the `KillOrigin`.
+		/// - `proposal_hash`: The hash of the proposal that should be killed.
+		///
+		/// Emits `Killed` and `ProposalCostBurned` if any cost was held for a given proposal.
+		#[pallet::call_index(7)]
+		#[pallet::weight(T::WeightInfo::kill(1, T::MaxProposals::get()))]
+		pub fn kill(origin: OriginFor<T>, proposal_hash: T::Hash) -> DispatchResultWithPostInfo {
+			T::KillOrigin::ensure_origin(origin)?;
+			let burned = if let Some((who, cost)) = <CostOf<T, I>>::take(proposal_hash) {
+				cost.burn(&who);
+				Self::deposit_event(Event::ProposalCostBurned { proposal_hash, who });
+				true
+			} else {
+				false
+			};
+			let proposal_count = Self::remove_proposal(proposal_hash);
```

Comment:
This does not check whether the proposal exists 🤔 ? Yes there is the consideration, but i would rather do an explicit check as well.

Reply:
updated